### PR TITLE
Column to list performance

### PR DESCRIPTION
**Pull request by jeffbrennan**
addresses #53 

Uses timeit module to repeat the operations described in https://mungingdata.com/pyspark/column-to-list-collect-tolocaliterator/ until the sum of runtimes exceeds 2 seconds.

Plot of results is below - based on this, flatmap/map are the preferable implementations for typical spark workloads
newplot (1)

@MrPowers

### CODE REVIEW

**Comment by MrPowers** [comment]
This is so cool.  The graph is nice and pretty!!

**Comment by jeffbrennan** [comment]
updated results - enabling arrow makes a huge difference for toPandas() runtimes
@MrPowers @SemyonSinchenko 

2023-11-28 column_to_list_benchmark

**Comment by jeffbrennan** [comment]
2023-11-28 column_to_list_benchmark_lineplot

**Comment by SemyonSinchenko** [comment]
@jeffbrennan JFYI, `pandas` and `pyarrow` are optional, not mandatory dependencies of `PySpark`

File: benchmarks/benchmark_column_performance.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    result = auto_timeit(stmt, setup, min_runtime_seconds)
+
+    summary = {
+        "test_name": test_name,
+        "dataset": dataset["name"],
+        "dataset_size": dataset["size"],
+        "runtimes": result,
+    }
+
+    with Path.open(f"benchmarks/results/{test_name}_{dataset['name']}.json", "w") as f:
```
[the remark this answers is not in the document]
Comment (jeffbrennan) [reply benchmarks/benchmark_column_performance.py]: yeah good point - thanks for noticing that!

**Comment by jeffbrennan** [comment]
> @jeffbrennan JFYI, `pandas` and `pyarrow` are optional, not mandatory dependencies of `PySpark`

yeah so it looks like flatmap is the best option to use as a default, with toPandas as an optional performance improvement. 

@MrPowers, @SemyonSinchenko  let me know what y'all think would be the best way to implement that. Something like checking the spark configuration and using the `toPandas` method if arrow optimization is enabled in the provided `SparkSession`?

**Comment by jeffbrennan** [comment]
@MrPowers @SemyonSinchenko

hey sorry for the delay in updates on this. I reran the benchmark using repartition(1) on smaller datasets and put the results into a jupyter notebook. Let me know if y'all have any feedback and if I need to do anything else for mkdocs to work.

**Pull request merged**